### PR TITLE
Add ability to generate a new ppt from a template and a list of updates

### LINKED ISSUE #1: Add ability to generate a new ppt from a template and a list of updates
#### Description:
The updates should be
- simple text in shape
- image
- chart data

### PR DESCRIPTION
**Pull request by adecler**
### NOTE: Depends on 
You need to have the latest version of the Excel adapter to be compiled.

### Issues addressed by this PR

Closes #1

See issue for details

### Test files

[PowerPointUpdate_Test.zip](https://github.com/BHoM/PowerPoint_Toolkit/files/10911952/PowerPointUpdate_Test.zip)
To push the data to the adapter, switch cell `B42` to `TRUE`

The initial slide 3 in the template file looks something like this :
image

and slide 3 from the output file should look like this:
image

Obviously, feel free to play with the input values. 

### Additional comments
A keen eye might have noticed that the colour purple is used for the last series as well as the first. That is not a bug. This is because there are more series in the new chart than in the original so I am reuse the colours in cycle.  This should ideally never happen with a well designed template but it was good to test.

### CODE REVIEW

File: PowerPoint_Adapter/PowerPoint_Adapter.csproj [hunk 1]
```diff
@@ -8,12 +8,23 @@
     <Description>https://github.com/BHoM/PowerPoint_Toolkit</Description>
     <Version>5.0.0</Version>
     <Authors>BHoM</Authors>
-    <Copyright>Copyright � https://github.com/BHoM</Copyright>
+    <Copyright>Copyright � https://github.com/BHoM</Copyright>
```
Comment (FraserGreenroyd) [inline PowerPoint_Adapter/PowerPoint_Adapter.csproj]: 👀

File: PowerPoint_Playground/PowerPoint_Playground.csproj [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  <ItemGroup>
+    <ProjectReference Include="..\PowerPoint_Adapter\PowerPoint_Adapter.csproj" />
+    <ProjectReference Include="..\PowerPoint_Engine\PowerPoint_Engine.csproj" />
+    <ProjectReference Include="..\PowerPoint_oM\PowerPoint_oM.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Reference Include="Adapter_oM">
+      <HintPath>..\..\..\ProgramData\BHoM\Assemblies\Adapter_oM.dll</HintPath>
```
Comment (FraserGreenroyd) [inline PowerPoint_Playground/PowerPoint_Playground.csproj]:
```suggestion
      <HintPath>C:\ProgramData\BHoM\Assemblies\Adapter_oM.dll</HintPath>
```

File: PowerPoint_Playground/PowerPoint_Playground.csproj [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    </Reference>
+    <Reference Include="BHoM_Adapter">
+      <HintPath>..\..\..\ProgramData\BHoM\Assemblies\BHoM_Adapter.dll</HintPath>
```
Comment (FraserGreenroyd) [inline PowerPoint_Playground/PowerPoint_Playground.csproj]:
```suggestion
      <HintPath>C:\ProgramData\BHoM\Assemblies\BHoM_Adapter.dll</HintPath>
```

File: PowerPoint_Playground/PowerPoint_Playground.csproj [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    </Reference>
+    <Reference Include="Excel_oM">
+      <HintPath>..\..\..\ProgramData\BHoM\Assemblies\Excel_oM.dll</HintPath>
```
Comment (FraserGreenroyd) [inline PowerPoint_Playground/PowerPoint_Playground.csproj]:
```suggestion
      <HintPath>C:\ProgramData\BHoM\Assemblies\Excel_oM.dll</HintPath>
```

File: PowerPoint_oM/PowerPoint_oM.csproj [hunk 5]
```diff
@@ -12,12 +12,23 @@
...
+  </PropertyGroup>
+
   <ItemGroup>
     <Reference Include="BHoM">
       <HintPath>C:\ProgramData\BHoM\Assemblies\BHoM.dll</HintPath>
       <Private>false</Private>
       <SpecificVersion>false</SpecificVersion>
     </Reference>
+    <Reference Include="Excel_oM">
+      <HintPath>..\..\..\ProgramData\BHoM\Assemblies\Excel_oM.dll</HintPath>
```
Comment (FraserGreenroyd) [inline PowerPoint_oM/PowerPoint_oM.csproj]:
```suggestion
      <HintPath>C:\ProgramData\BHoM\Assemblies\Excel_oM.dll</HintPath>
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: PowerPoint_Adapter/PowerPoint_Adapter.csproj (hunk 1 above)
Comment (adecler) [reply PowerPoint_Adapter/PowerPoint_Adapter.csproj]: How did that even happened ??
Fixed

File: PowerPoint_Playground/PowerPoint_Playground.csproj (hunk 2 above)
Comment (adecler) [reply PowerPoint_Playground/PowerPoint_Playground.csproj]: That project is not supposed to be there anymore.
 File deleted

File: PowerPoint_Playground/PowerPoint_Playground.csproj (hunk 3 above)
Comment (adecler) [reply PowerPoint_Playground/PowerPoint_Playground.csproj]: That project is not supposed to be there anymore.
 File deleted

File: PowerPoint_Playground/PowerPoint_Playground.csproj (hunk 4 above)
Comment (adecler) [reply PowerPoint_Playground/PowerPoint_Playground.csproj]: That project is not supposed to be there anymore.
 File deleted

File: PowerPoint_oM/PowerPoint_oM.csproj (hunk 5 above)
Comment (adecler) [reply PowerPoint_oM/PowerPoint_oM.csproj]: I always miss at least one!
Fixed

File: PowerPoint_Adapter/PowerPoint_Adapter.csproj (hunk 1 above)
Comment (FraserGreenroyd) [reply PowerPoint_Adapter/PowerPoint_Adapter.csproj]: 🤷‍♂️ first time I've ever seen it! 😄

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix copyright headers ref. 11931380534

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix project file ref. 11931376330

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Review by FraserGreenroyd — APPROVED** [review]
I downloaded today's alpha and then built this PR branch on top of that alpha.
I downloaded the test script saved in the ZIP folder on this PR.
I ran the Excel script per the instructions and got the results which are described in the PR. I then deleted the PPT output, and regenerated it and got a new PPT back with the expected slide changes from the template. I then changed a slide title and got that output visible on the PPT as well. I am happy with the functionality as described.

I have reviewed the code alongside BHoMBot's checks and am happy with the code.

One minor thing to note, the PPT output could not be opened by PowerPoint on every occasion without closing Excel first (nor could it be deleted stating that Excel had ownership of the file). Closing Excel resolved both problems, and opening the PPT revealed the expected output, so I'm considering it a minor issue that may need investigating later should others find it to be a problem, but while this toolkit is in early development mode, I don't think it's a problem so I'm approving and merging this PR.

**Comment by adecler** [comment]
Thanks @FraserGreenroyd , I have raised a separate issue here: https://github.com/BHoM/PowerPoint_Toolkit/issues/3

**Pull request merged**